insights/combiners/limits_conf.py

```diff
@@ -0,0 +1,75 @@
+"""
+Limits configuration
+====================
+
+The limits files are normally available to rules as a list of LimitsConf
+objects.  This combiner turns those into one set of data, and provides a
+``find_all()`` method to search the rules from all the files.
+"""
+
+from insights.core.plugins import combiner
+from insights.parsers.limits_conf import LimitsConf
+
+
+@combiner(requires=[LimitsConf])
+class AllLimitsConf(object):
+    """
+    Combiner for accessing all the limits configuration files.
+
+    Attributes:
+        domains(set): the set of domains found in all data files.
+        limits(list): a list of the original LimitsConf parser results.
+        rules(list): the entire list of rules.
+    """
+    def __init__(self, local, shared):
+        rules = []
+        domains = set([])
+        self.limits = shared[LimitsConf]
+        # Store the tuple of domain, type and item against the row of that
+        # limit in our full list of rules.
+        rule_tuples_found = {}
+        rule_number = 0
+        for limits in shared[LimitsConf]:
```

Comment:
Hi @chenlizhong - from what I can see the items in shared[LimitsConf] will automatically be in file name order by spec, so limits.conf will be listed first and then all the files in limits.d in name order.  This is because they are read into the spec handling system in name order.